packages/squiggle-lang/src/rescript/Utility/E/E_A.res

```diff
@@ -323,32 +323,45 @@ module Floats = {
       let continuous: array<float> = []
       let discrete = FloatFloatMap.empty()
 
-      let addData = (count: int, value: float): unit => {
-        if count >= minDiscreteWeight {
-          FloatFloatMap.add(value, count->Belt.Int.toFloat, discrete)
+      // In a run of exactly minDiscreteWeight, the first and last
+      // element indices differ by minDistance.
+      let minDistance = minDiscreteWeight - 1
+
+      let len = length(sortedArray)
+      let i = ref(0)
+      while i.contents < len - minDistance {
+        let value = sortedArray[i.contents]
+        if value != sortedArray[i.contents + minDistance] {
+          Js.Array2.push(continuous, value)->ignore
```

Comment:
Just to be clear, I assume that we're assuming that writing a function to wrap this, like,
`let addValuesToContinuousData = (value) =>  Js.Array2.push(continuous, value)->ignore`
would suffer a performance penalty? I know @berekuk mentioned that function calls are expensive, but I'm not sure on the specifics. I would have really expected the V8 engine to optimize this.

Reply:
I don't think it would be slower, I just don't find it any clearer.